modyn/supervisor/internal/triggers/performancetrigger.py

```diff
@@ -0,0 +1,302 @@
+                        mode=self.config.mode,
+                        method=self.config.forecasting_method,
+                    )
+                    if policy_decisions[policy_name]:
+                        any_policy_triggered = True
+
+                triggered = any_policy_triggered
+
+            if triggered:
+                for policy in self.decision_policies.values():
+                    policy.inform_trigger()  # resets the internal state (e.g. misclassification counters)
+
+            trigger_idx = processing_head_in_batch - 1
+
+            drift_eval_log = PerformanceTriggerEvalLog(
+                triggered=triggered,
+                trigger_index=trigger_idx,
+                evaluation_interval=(
+                    next_detection_interval[0][1],
+                    next_detection_interval[-1][1],
+                ),
+                num_samples=num_samples,
+                num_misclassifications=num_misclassifications,
+                evaluation_scores=evaluation_scores,
+                policy_decisions=policy_decisions,
+            )
+            if log:
+                log.evaluations.append(drift_eval_log)
+
+            self._last_detection_interval = next_detection_interval
+            if triggered:
+                yield trigger_idx
+
+    def inform_new_model(self, most_recent_model_id: int) -> None:
+        self.most_recent_model_id = most_recent_model_id
+        self.model_refresh_needed = True
+
+        # Perform an evaluation of the NEW model on the last evaluation interval, we will derive expected performance
+        # forecasts from these evaluations.
+        num_samples, num_misclassifications, evaluation_scores = self._run_evaluation(
+            interval_data=self._last_detection_interval
+        )
+
+        self.performance_tracker.inform_trigger(
+            num_samples=num_samples,
+            num_misclassifications=num_misclassifications,
+            evaluation_scores=evaluation_scores,
+        )
+
+    # ---------------------------------------------------------------------------------------------------------------- #
+    #                                                     Internal                                                     #
+    # ---------------------------------------------------------------------------------------------------------------- #
+
+    def _run_evaluation(self, interval_data: list[tuple[int, int]]) -> tuple[int, int, dict[str, float]]:
+        """Run the evaluation on the given interval data."""
+        assert self.most_recent_model_id is not None
+        assert self.dataloader_info is not None
+        assert self.model_downloader is not None
+        assert self.context and self.context.pipeline_config is not None
+
+        evaluation_dataloader = prepare_trigger_dataloader_fixed_keys(
+            self.dataloader_info, [key for key, _ in interval_data]
+        )
+
+        # Download most recent model as model manager
```

Comment:
nit: manager has been renamed

Reply:
changes from earlier are not merged in, yet